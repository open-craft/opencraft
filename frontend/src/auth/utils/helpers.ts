import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
