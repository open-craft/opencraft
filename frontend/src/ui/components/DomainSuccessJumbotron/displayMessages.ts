const messages = {
  domainIsAvailable: {
    defaultMessage: 'Your domain name is available',
    description: 'Highlighted message telling user their domain is available.'
  },
  domainIsConnected: {
    defaultMessage: 'Your domain name is connected',
    description: 'Highlighted message telling user their domain is available.'
  },
  secureDomainNow: {
    defaultMessage:
      'To secure your domain now, please fill in the details below.'
  }
};

export default messages;
