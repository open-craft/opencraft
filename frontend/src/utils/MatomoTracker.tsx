import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
