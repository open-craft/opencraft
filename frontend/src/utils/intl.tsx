import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
