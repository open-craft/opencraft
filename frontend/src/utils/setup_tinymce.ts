import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
