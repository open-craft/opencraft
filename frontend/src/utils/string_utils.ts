export const toCamelCase = (s: string) => {
  return s.replace(/([-_][a-z])/gi, ($1) => {
    return $1.toUpperCase().replace('-', '').replace('_', '');
  });
};

export const sanitizeErrorFeedback = (input: {
  [key: string]: Array<string>;
}) => {
  // Loop at each error message and join them.
  // Also convert keys from snake_case to camelCase
  const newObject: any = {};

  Object.keys(input).forEach((key) => {
    newObject[toCamelCase(key)] = input[key].join();
  });

  return newObject;
};

export const capitalizeFirstLetter = (str: string) => {
  return str[0].toUpperCase() + str.slice(1);
};
