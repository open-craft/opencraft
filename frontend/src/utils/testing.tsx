import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
