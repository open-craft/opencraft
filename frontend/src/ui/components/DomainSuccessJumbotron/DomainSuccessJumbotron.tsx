import * as React from 'react';
import { Jumbotron } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import { INTERNAL_DOMAIN_NAME } from 'global/constants';
import iconCheck from 'assets/circle-check.png';
import messages from './displayMessages';
import './styles.scss';

interface DomainProps {
  domain: string;
  domainIsExternal: boolean;
}

export const DomainSuccessJumbotron: React.FC<DomainProps> = (
  props: DomainProps
) => {
  let domainStatusText: string;

  if (props.domainIsExternal) {
    domainStatusText = 'domainIsConnected';
  } else {
    domainStatusText = 'domainIsAvailable';
  }

  return (
    <Jumbotron className="domain-available">
      <img src={iconCheck} alt="" />
      <h2>
        <WrappedMessage id={domainStatusText} messages={messages} />
      </h2>
      <div className="domain-name">
        <p>
          {props.domain}
          {!props.domainIsExternal && <span>{INTERNAL_DOMAIN_NAME}</span>}
        </p>
      </div>
      <p>
        <WrappedMessage id="secureDomainNow" messages={messages} />
      </p>
    </Jumbotron>
  );
};
