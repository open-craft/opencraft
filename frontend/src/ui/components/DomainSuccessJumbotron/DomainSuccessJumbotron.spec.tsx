import React from 'react';
import { shallow } from 'enzyme';
import { setupComponentForTesting } from "utils/testing";
import { DomainSuccessJumbotron } from './DomainSuccessJumbotron';

it('renders without crashing', () => {
  const tree = setupComponentForTesting(<DomainSuccessJumbotron />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correct text when using external domain', () => {
  const tree = setupComponentForTesting(
    <DomainSuccessJumbotron domain="www.test.com" domainIsExternal={true} />
  ).toJSON();
  expect(tree.children[1].children[0]).toEqual('Your domain name is connected');
  expect(tree.children[2].children[0].children[0]).toEqual('www.test.com');
});

it('renders correct text when using internal domain', () => {
  const tree = setupComponentForTesting(
    <DomainSuccessJumbotron domain="www.test.com" domainIsExternal={false} />
  ).toJSON();
  expect(tree.children[1].children[0]).toEqual('Your domain name is available');
  expect(tree.children[2].children[0].children[0]).toEqual('www.test.com');
});
