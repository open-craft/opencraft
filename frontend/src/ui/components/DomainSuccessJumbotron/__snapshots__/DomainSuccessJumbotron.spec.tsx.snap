// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="domain-available jumbotron"
>
  <img
    alt=""
    src="circle-check.png"
  />
  <h2>
    Your domain name is available
  </h2>
  <div
    className="domain-name"
  >
    <p />
  </div>
  <p>
    To secure your domain now, please fill in the details below.
  </p>
</div>
`;
